Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py

```diff
@@ -125,31 +125,30 @@ def _append(branch, hash, revision=None):
             intersected = False
             log = None
             try:
-                kwargs = dict()
-                if sys.version_info >= (3, 6):
-                    kwargs = dict(encoding='utf-8')
                 self._last_populated[branch] = time.time()
                 log = subprocess.Popen(
                     [self.repo.executable(), 'log', '{}/{}'.format(remote, branch) if remote else branch, '--no-decorate', '--date=unix', '--'],
                     cwd=self.repo.root_path,
                     stdout=subprocess.PIPE,
                     stderr=subprocess.PIPE,
-                    **kwargs
                 )
                 if log.poll():
                     raise self.repo.Exception("Failed to construct branch history for '{}'".format(branch))
 
                 hash = None
                 revision = None
 
-                line = log.stdout.readline()
+                import codecs
+                log_stdout = codecs.getreader('ascii')(log.stdout, 'backslashreplace')
```

Comment:
I'm a bit confused why this failure is unique to Python 3.10? Seems like many versions of Python would suffer from this, no?

Reply:
Good point. I built Python 3.10.6 for the sake of testing and worked fine, so I am going to guess it has something to do with how Buildroot configures its host tools (Python, Git, or maybe even others?)

I suppose we would still want to make things robust to odd content in commit logs, though?

Reply:
That's a fair point, and in this case, `ascii` is correct.

I'm going to hold off on approving and landing until tomorrow morning, though, because this is the kind of change that could effect CI in surprising ways.